react-effects/my-app/src/List.js

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable no-unused-vars -- REMOVE ME */
+import { useEffect, useState } from 'react';
+import readItems from './read';
+
+export default function List() {
+  const [isLoading, setIsLoading] = useState(false);
+  const [items, setItems] = useState([]);
+  const [error, setError] = useState();
+
+  // Your code here:
+  //  - When the component mounts:
+  //    - Read the items using `readItems` and update state so the list displays
+  //    - Handle errors from `readItems`
+  useEffect(() => {
+    async function loadItems() {
+      try {
+        const items = await readItems();
+        setItems(items);
+        setIsLoading(false);
+      } catch (error) {
+        setError(error);
+        setIsLoading(false);
+      }
+    }
+    loadItems();
+  }, [isLoading])
```

Comment:
With your code as it now stands, `isLoading` is not a dependency since the Effect function does not reference `isLoading` directly.

With `isLoading` listed as a dependency, this effect will re-run every time `isLoading` changes. That means that after you issue the `readItems` and call `setIsLoading(false)`, then `isLoading` changes (from true to false). That will cause this effect to run again, which will result in another call to `readItems`. So, I'm guessing you are seeing 4 calls to `readItems` instead of just 2 when you refresh your page.

Removing `isLoading` as a dependency will leave you with just `[]`, which means it will run only on first mount. In that case, you would see just 2 calls to `readItems` (one from the call when it's mounted and one from the StrictMode extra call).

Reply:
Okay, will fix that.  Thank you.